File: app/src/main/java/com/hjc/base/ui/home/page/ConcertFactory.java
package com.hjc.base.ui.home.page;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;

import com.hjc.base.bean.ConcertBean;


public class ConcertFactory extends DataSource.Factory<Integer, ConcertBean> {
    private MutableLiveData<ConcertDataSource> mSourceData = new MutableLiveData<>();

    @NonNull
    @Override
    public DataSource<Integer, ConcertBean> create() {
        ConcertDataSource source = new ConcertDataSource();
        mSourceData.postValue(source);
        return source;
    }
}

File: module-home/src/main/java/com/hjc/module_home/ui/paging/ConcertFactory.kt
package com.hjc.module_home.ui.paging

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import com.hjc.module_home.entity.ConcertBean

class ConcertFactory : DataSource.Factory<Int, ConcertBean>() {

    private val mSourceData: MutableLiveData<ConcertDataSource> =
        MutableLiveData<ConcertDataSource>()

    override fun create(): DataSource<Int, ConcertBean> {
        val source = ConcertDataSource()
        mSourceData.postValue(source)
        return source as DataSource<Int, ConcertBean>
    }
}